### ISSUE TITLE
Installation using Ubuntu software center fails on 14.04.3 LTS

### ISSUE THREAD

**Issue by EdDixon** [learningequality/installers#145]
While testing the installation procedures on various OS's, I encountered an issue with using the Ubuntu Software Center and Ubuntu LTS. Replicated by simply double clicking the current software package ka-lite-bundle_0.14~b3-0ubuntu11_all.deb from within the nautilus file explorer which will open the package using the software center. The progress bar remains unmoved for a long time then suddenly fills to the right but stops just short of completion and the software center windows turns dark. I have allowed sometime for this to resolve on its own  knowing it is a fairly large package. Closing the windows provides only the option to force quit which afterwards the software is treated as if installed and a kalite command of any kind reveals that it is trying to complete the install but needs to know which user to use the command with. The two screen shots added here are 1) the results of issuing a kalite command after encountering the problem same results happen in terminal when force quitting: 
![lts-2-2015-08-24_17-54-12](https://cloud.githubusercontent.com/assets/4349806/9486112/27465e54-4b80-11e5-8e3a-4c682986b4be.png)
and 2) state of the frozen software center window when encountering this problem:
![lts-1-2015-08-24_17-35-44](https://cloud.githubusercontent.com/assets/4349806/9486106/1ababb12-4b80-11e5-9d74-34b0a3e25a24.png)

**Comment by EdDixon** [comment]
I do know that 14.03.4 LTS has a slightly lower than recommended python version (warning issued during a manually dpkg -i install) and will check and see if using the newer 15.04 resolves the issue (which I believe does have a newer python version tomorrow?

**Comment by radinamatic** [comment]
@EdDixon Are you certain you deleted all the trace of previous installations you may have had on that machine? 

The only issue I had while installing through Ubuntu Software Center on 15.04, was that **dbconf** window opened behind and was waiting for confirmation weather to install on system startup. Once it received that confirmation, the installation appeared as completed. In my case, the ″default user selection″ from the Terminal in your first screenshot appeared as a part of `kalite manage setup` command that needs to be executed after the installation inside Ubuntu Software Center has finished.

**Comment by EdDixon** [comment]
Yes absolutely sure there had never been an install of ka-lite on that drive before. In fact this was tested many times trying to resolve the issue here. There was no debconf window ever during the test I tried minimizing all windows etc just to be sure during the installation and after the fail? I was unaware that there is a need to run kalite manage setup after installation though, which in that case sounds like it may have fully installed despite the need to force quit in the software center after installation? Can you remember what information the debconf requires the user to provide? I would think choosing the default user would be there? I do plan to test this same install on 15.04 tomorrow so I will see soon enough myself anyway but only the window shown n the first screenshot was provided after double clicking on the package?

**Comment by EdDixon** [comment]
Regretfully the 15.04 installation using the software center has also failed for me in exactly the same way as the 14 LTS. @MCGallaspy has already adjusted the documentation accordingly to exclude this method of installation and moved it to a 15 milestone so we will be getting back to this soon!

**Comment by benjaoming** [comment]
@MCGallaspy any reason for the 0.15.0 milestone?

**Comment by benjaoming** [comment]
Sorry, didn't see @EdDixon's last comment there. I'll see if there's a simple explanation for this, since we're still on the last beta, and I can just make another build in case there's a reasonable fix.

**Comment by MCGallaspy** [comment]
@benjaoming so that we don't hold up 0.14 release if possible

**Comment by benjaoming** [comment]
This might be related... output from Software Center after 13 minutes on my computer:

```
2015-08-26 23:01:56,930 - softwarecenter.backend - WARNING - _on_trans_error: 'DBusException(dbus.String(u'Message did not receive a reply (timeout by message bus)'),)' '<AptTransaction object at 0x7f2b28b91690 (aptdaemon+client+AptTransaction at 0x233ece0)>' 'ka-lite-bundle'
2015-08-26 23:01:56,955 - softwarecenter.backend - WARNING - daemon dies, ignoring: <AptTransaction object at 0x7f2b28b91690 (aptdaemon+client+AptTransaction at 0x233ece0)> exit-failed
```

**Comment by benjaoming** [comment]
Confirming this for ka-lite-bundle on 14.04

However, ka-lite works fine.

**Comment by benjaoming** [comment]
@EdDixon @radinamatic I'm not able to reproduce this in the following scenario:

1. Add PPA
2. Open Software Center
3. Install ka-lite-bundle

I've tried both to remove and purge before installing. So everything I've tried so far worked... *with the PPA*.

So the issue is somehow related to installing the .deb directly.

Can you confirm this?

**Comment by EdDixon** [comment]
@benjaoming  I was under the impression the goal was to not have to add the PPA to install with the software center? My understanding was that it would just be downloaded and double clicked all in GUI (to not have to be on-line or in shell during the installation which I think adding the PPA may require?) I am happy to test that ASAP here. I do believe that may resolve it as well. The PPA installation seemed to work great in testing here. Is that the typical method used to install packages in Ubuntu using PPA's though or can the PPA information be somehow added to the package and installed into apt offline before running the rest of the package installation? Its obvious how much more experience I have with Debian than Ubuntu at this point I guess?

**Comment by radinamatic** [comment]
@benjaoming @EdDixon 

My test-case was an **offline** installation (WiFi turned off) from previously downloaded (directly from Launchpad) `ka-lite-bundle_0.14~b3-0ubuntu11_all.deb` package, as per @jamalex indication, aiming to review if that package installed all the assessment items correctly (and it does) when offline. To my understanding PPA is not a factor here as it only allows user to access & download from a specific source, but should have no bearing on the installation process itself.

I removed and purged all the previous installations, but had to manually delete /etc/ka-lite and ~/.kalite directories since `dpkg purge` does not do it automatically. The only issue I encountered (on 15.04) was the `dbconf` window one described above.
